### ISSUE TITLE
[UX] Boba Gateway/Bridge: Subtract gas fees automatically

### ISSUE THREAD

**Issue by wsdt** [omgnetwork/optimism-v2#453]
Thought I would just drop this here. When bridging ETH to Boba (Rinkeby in this case) then the transaction would fail when trying to transfer the full amount. 

What I've seen for some other bridges etc. is that they automatically subtract the gas from the amount to transfer which makes the bridge way more user-friendly. 

Some suggestions: 

- Directly subtract the gas fee from within the smart contract
- Just subtract the estimated gas fee client-side (+ some buffer maybe)
- ...

![image](https://user-images.githubusercontent.com/28724551/156769291-4f45d01f-e99f-473d-a779-c93e2fbe0f99.png)

Suggestions? Feedback?

**Comment by CAPtheorem** [comment]
Yes we had that feature ("Bridge All") but removed it because it was error prone. One issue was that users would select "Bridge All", but then wait for several minutes before signing the transaction. If the gas increased in that time window, the transaction would then fail, and conversely, if the gas decreased in that delay window, the user would have funds left in their account. Given that ETH is the fee token, it is particularity difficult to get "Bridge All" to work reliability in all circumstances.
